tests/integration/test_schema_avro_references.py

```diff
@@ -0,0 +1,295 @@
+    "name": "Country",
+    "namespace": "com.netapp",
+    "fields": [{"name": "name", "type": "string"}, {"name": "code", "type": "string"}],
+}
+
+# address.avsc
+SCHEMA_ADDRESS = {
+    "type": "record",
+    "name": "Address",
+    "namespace": "com.netapp",
+    "fields": [
+        {"name": "street", "type": "string"},
+        {"name": "city", "type": "string"},
+        {"name": "postalCode", "type": "string"},
+        {"name": "country", "type": "Country"},
+    ],
+}
+
+# job.avsc
+SCHEMA_JOB = {
+    "type": "record",
+    "name": "Job",
+    "namespace": "com.netapp",
+    "fields": [{"name": "title", "type": "string"}, {"name": "salary", "type": "double"}],
+}
+
+# person.avsc
+SCHEMA_PERSON = {
+    "type": "record",
+    "name": "Person",
+    "namespace": "com.netapp",
+    "fields": [
+        {"name": "name", "type": "string"},
+        {"name": "age", "type": "int"},
+        {"name": "address", "type": "Address"},
+        {"name": "job", "type": "Job"},
+    ],
+}
+
+SCHEMA_PERSON_INT_LONG = {
+    "type": "record",
+    "name": "Person",
+    "namespace": "com.netapp",
+    "fields": [
+        {"name": "name", "type": "string"},
+        {"name": "age", "type": "long"},
+        {"name": "address", "type": "Address"},
+        {"name": "job", "type": "Job"},
+    ],
+}
+
+SCHEMA_PERSON_INT_STRING = {
+    "type": "record",
+    "name": "Person",
+    "namespace": "com.netapp",
+    "fields": [
+        {"name": "name", "type": "string"},
+        {"name": "age", "type": "string"},
+        {"name": "address", "type": "Address"},
+        {"name": "job", "type": "Job"},
+    ],
+}
+
+SCHEMA_UNION_REFERENCES = {
+    "type": "record",
+    "namespace": "com.netapp",
+    "name": "Person2",
+    "fields": [
+        {"name": "name", "type": "string"},
+        {"name": "age", "type": "int"},
+        {"name": "address", "type": "Address"},
+        {"name": "job", "type": "Job"},
+        {
+            "name": "children",
+            "type": [
+                "null",
+                {
+                    "type": "record",
+                    "name": "child",
+                    "fields": [{"name": "name", "type": "string"}, {"name": "age", "type": "int"}],
+                },
+            ],
+        },
+    ],
+}
+
+SCHEMA_UNION_REFERENCES2 = [
+    {
+        "type": "record",
+        "name": "Person",
+        "namespace": "com.netapp",
+        "fields": [
+            {"name": "name", "type": "string"},
+            {"name": "age", "type": "int"},
+            {"name": "address", "type": "Address"},
+            {"name": "job", "type": "Job"},
+        ],
+    },
+    {
+        "type": "record",
+        "name": "UnemployedPerson",
+        "namespace": "com.netapp",
+        "fields": [
+            {"name": "name", "type": "string"},
+            {"name": "age", "type": "int"},
+            {"name": "address", "type": "Address"},
+        ],
+    },
+]
+
+SCHEMA_ADDRESS_INCOMPATIBLE = {
```

Comment:
I cannot find the incompatibility in this schema.

Reply:
This test shows incompatibility in the naming of the field... We changed the name from "Address" to "ChangeAddress".